### ISSUE TITLE
Implement AST Generation for next-node prediction and text completion

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#692]
## Goal
Enable Klareco to generate text via **next-AST-node prediction** instead of next-token prediction, maintaining grammatical correctness by construction.

## Motivation
Traditional LLMs predict text tokens, which allows grammatical errors. Klareco should predict **AST nodes**, then deparse to text, ensuring 100% grammatical validity.

### ASTGenerator Model (~20M params)
```python
class ASTGenerator(nn.Module):
    """Predicts next AST node given partial AST context"""
    
    def __init__(self, embedding_dim=128, hidden_dim=256):
        self.tree_encoder = TreeLSTMEncoder(...)  # Encode partial AST
        self.node_predictor = nn.Sequential(
            nn.Linear(hidden_dim, len(VORTSPECOJ)),  # Predict word type
            nn.Linear(hidden_dim, len(ROOT_VOCAB)),   # Predict root
            nn.Linear(hidden_dim, len(KAZOJ)),        # Predict case
            # ... other grammatical features
        )
    
    def forward(self, partial_ast):
        context = self.tree_encoder(partial_ast)
        next_node = self.node_predictor(context)
        return next_node
```

### Training Data Preparation
1. Parse corpus to ASTs (already have in unified_corpus.jsonl)
2. Create training pairs: (partial AST → next node)
3. Loss: cross-entropy on node predictions

### Generation Pipeline
```
User input: "Mi amas la hun..."
  ↓
Parse partial input → partial AST
  ↓
ASTGenerator predicts next node → {radiko: "hund", vortspeco: "substantivo", kazo: "akuzativo"}
  ↓
Deparse to text → "hundon"
```

## Success Criteria
- [ ] ASTGenerator model implemented and tested
- [ ] Training script: `scripts/train_ast_generator.sh`
- [ ] Generate grammatically valid completions (100% parse rate)
- [ ] Evaluation: coherence and relevance on held-out corpus
- [ ] Integration with CLI: `klareco generate "Mi amas la hun..." --complete`

## Dependencies
- Stage 1: Root embeddings (#685)
- Stage 2: Compositional embeddings (in progress)
- TreeLSTMEncoder (already implemented)

## Related Issues
- #690: Abstractive answer generation (uses AST generation)
- #680: Extractive answer generation (current approach)

## Training Pipeline Position
```
Root Embeddings → Compositional Embeddings → [AST Generator] → Generation Tasks
```

## Implementation Notes
- Key advantage: Grammatically invalid completions are **impossible**
- Can be used for: text completion, paraphrasing, question generation
- Smaller model than traditional LLMs (~20M vs 70B+ params)

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
